pkg/zonemgr/zone_file_cache.go

```diff
@@ -55,10 +56,33 @@ func (zoneFileCache *ZoneFileCache) prepare() {
 	zoneFileCache.generateHeaderSuffix()
 	zoneFileCache.soaSerial = 0
 	zoneFileCache.header = zoneFileCache.generateHeader()
-	zoneFileCache.content = zoneFileCache.header
 	zoneFileCache.vmiRecordsMap = make(map[string][]string)
 }
 
+func (zoneFileCache *ZoneFileCache) updateIfAlreadyExist(content []byte) {
+	if prevSoaSerial := fetchSoaSerial(content); prevSoaSerial > 0 {
+		zoneFileCache.soaSerial = prevSoaSerial + 1
+		zoneFileCache.header = zoneFileCache.generateHeader()
+	}
+}
+
+func fetchSoaSerial(content []byte) int {
+	contentStr := string(content)
+	var ind1, ind2 int
+	if ind1 = strings.Index(contentStr, "("); ind1 == 0 {
+		return 0
+	}
+	if ind2 = strings.Index(contentStr[ind1:], " "); ind2 == 0 {
+		return 0
+	}
+	soaSerial := strings.TrimSpace(contentStr[ind1+1 : ind1+ind2])
+	if soaSerialInt, err := strconv.Atoi(soaSerial); err != nil {
+		return 0
+	} else {
+		return soaSerialInt
+	}
+}
+
```

Comment:
there might be a way using regex to fetch it, will be nice if we can check please

Reply:
Btw according https://www.ietf.org/rfc/rfc1035.txt you can add comments (start with `;`)
if the first line will be a comment with the serial, you can read just that line instead the whole file and get the serial

Reply:
Changed to using regex.
Regarding the "reading line instead of the whole file", I can add it in a different PR